File: app/src/main/java/com/example/wilfred/journalapp/database/EntryViewModel.java
package com.example.wilfred.journalapp.database;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

/**
 * ViewModel that will let app data survive connfigurations changes
 */

public class EntryViewModel extends AndroidViewModel {

    private EntryRepository mRepository;
    private LiveData<List<JournalEntry>> mAllEntries;

    public EntryViewModel(@NonNull Application application) {
        super(application);
        this.mRepository = new EntryRepository(application);
        mAllEntries = this.mRepository.getEntries();
    }

    public void insert(JournalEntry entry){
        mRepository.insert(entry);
    }
    public LiveData<List<JournalEntry>> getAllEntries() {
        if (mAllEntries == null){
           mAllEntries =  mRepository.getEntries();
        }
        return mAllEntries;
    }
    public void delete(JournalEntry entry){
        mRepository.delete(entry);
    }


}

File: app/src/main/java/com/example/wilfred/journalapp/database/EntryViewModel.kt
package com.example.wilfred.journalapp.database

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

/**
 * ViewModel that will let app data survive connfigurations changes
 */

class EntryViewModel(application: Application) : AndroidViewModel(application) {

    private val mRepository: EntryRepository = EntryRepository(application)
    private var mAllEntries: LiveData<List<JournalEntry>>? = null

    val allEntries: LiveData<List<JournalEntry>>?
        get() {
            if (mAllEntries == null) {
                mAllEntries = mRepository.entries
            }
            return mAllEntries
        }

    init {
        mAllEntries = this.mRepository.entries
    }

    fun insert(entry: JournalEntry) {
        mRepository.insert(entry)
    }

    fun delete(entry: JournalEntry) {
        mRepository.delete(entry)
    }

    fun getEntryById(id: Int): LiveData<JournalEntry> {
        return mRepository.getEntryById(id)
    }


    fun update(entry: JournalEntry) {
        mRepository.updateJournal(entry)
    }
}
